plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java

```diff
@@ -214,17 +215,19 @@ private void schedule(@Nullable DocumentEvent event) {
             }
 
             // TODO should we really schedule a new task each time this method called? or maybe at least cancel it at first
-            this.task = new TimerTask() {
+            this.task = new AbstractJob("Background parsing job") {
```

Comment:
Have you considered marking it as a system job so it won't be shown in the background tasks?

Reply:
Actually no. I want it to be shown as a separate background task to have ability to cancel it. I even added subtasks with progress. So, it user open a really big query or connection is slow, it's shown in the Background tasks - you see that something is happening, you see some progress.